### PR TITLE
Add `description` field in CandidateState

### PR DESCRIPTION
**Pull request by Tommo-L**
Since neo3 governance need to vote the committees and have a new economic model, more exchanges, wallets, mining pool will participate, they may need to know  the information about the candidates.

**Where in the software does this update applies to?**
- neo-cli
- sdk

### CODE REVIEW

File: src/neo/SmartContract/Native/Tokens/NeoToken.cs [hunk 1]
```diff
@@ -95,7 +95,7 @@ internal override bool Initialize(ApplicationEngine engine)
             for (int i = 0; i < Blockchain.CommitteeMembersCount; i++)
             {
                 ECPoint pubkey = Blockchain.StandbyCommittee[i];
-                RegisterCandidate(engine.Snapshot, pubkey);
+                RegisterCandidate(engine.Snapshot, pubkey, "Backup candidate");
```
Comment (shargon) [inline src/neo/SmartContract/Native/Tokens/NeoToken.cs:98]:
```suggestion
                RegisterCandidate(engine.Snapshot, pubkey, "Genesis candidate");
```

**Comment by erikzhang** [comment]
The description should not be on-chain.

**Comment by Tommo-L** [comment]
> The description should not be on-chain.

But, how wallets, exchanges, pos staking pools to support for voting without these information, or they could get the candidates information from neo.org, foundation or by neoid?

**Comment by shargon** [comment]
I think that it's good to have this information onchain, it's easier for all

**Comment by erikzhang** [comment]
But the information can be faked. In the end everyone still finds this information through the website.

**Comment by vncoelho** [comment]
I agree with both sides. The idea of publishing this info onchain is good.
I believe it should be optional and easy to update.

File: src/neo/SmartContract/Native/Tokens/NeoToken.cs [hunk 2]
```diff
@@ -95,7 +95,7 @@ internal override bool Initialize(ApplicationEngine engine)
             for (int i = 0; i < Blockchain.CommitteeMembersCount; i++)
             {
                 ECPoint pubkey = Blockchain.StandbyCommittee[i];
-                RegisterCandidate(engine.Snapshot, pubkey);
+                RegisterCandidate(engine.Snapshot, pubkey, "Genesis candidate");
```
Comment (vncoelho) [inline src/neo/SmartContract/Native/Tokens/NeoToken.cs:98]: In particular, I like this as well.

But perhaps the whole description could be optional.
If not provided we could just write "Description not provided".
And perhaps it needs an easy way to update.

**Pull request closed without merging**